src/AssetForm/containers/DraftInput.test.js

```diff
@@ -0,0 +1,240 @@
+        store = storeWithDraft(draft);
+        testInstance = render(
+          <DraftTextInput component={MockTextInput} name="testField" />,
+          { store }
+        );
+      });
+
+      test('sets value prop to empty string', () => {
+        expect(testInstance.findByType(MockTextInput).props.value).toBe('');
+      });
+    });
+  });
+
+  describe('with custom mapDraftToInputProps', () => {
+    let mapDraftToInputProps;
+
+    beforeEach(() => {
+      draft = { testField: 'x', other: 'y' };
+      store = storeWithDraft(draft);
+      mapDraftToInputProps = jest.fn(draft => ({ other: '>' + draft.other + '<' }));
+      testInstance = render(
+        <DraftTextInput
+          component={MockTextInput} name="testField"
+          mapDraftToInputProps={mapDraftToInputProps}
+        />, { store }
+      );
+    });
+
+    test('sets computed prop', () => {
+      expect(mapDraftToInputProps).toHaveBeenCalled();
+      expect(mapDraftToInputProps.mock.calls[0][0]).toBe(draft);
+      expect(testInstance.findByType(MockTextInput).props.other).toBe('>y<');
+    });
+  });
+
+  describe('with custom mapOnChangeToPatch', () => {
+    let mapOnChangeToPatch;
+
+    beforeEach(() => {
+      draft = { testField: 'x' };
+      store = storeWithDraft(draft);
+      mapOnChangeToPatch = jest.fn(
+        ({ name }, { target: { value } }) => ({ [name]: '>' + value + '<' })
+      );
+      testInstance = render(
+        <DraftTextInput
+          component={MockTextInput} name="testField"
+          mapOnChangeToPatch={mapOnChangeToPatch}
+        />, { store }
+      );
+    });
+
+    test('sets draft field by calling mapOnChangeToPatch', () => {
+      const input = testInstance.findByType(MockTextInput).instance;
+      input.setValue('x');
+      expect(mapOnChangeToPatch).toHaveBeenCalled();
+      expect(patchDraft).toHaveBeenCalledWith({ testField: '>x<' });
+    });
+  });
+});
+
+describe('DraftCheckboxInput', () => {
+  let draft, store, testInstance;
+
+  test('can render', () => {
+    testInstance = render(<DraftCheckboxInput name="testField" />);
+  });
+
+  describe('with a boolean field', () => {
+    [true, false].forEach(testValue => {
+      describe('with value of ' + testValue, () => {
+        beforeEach(() => {
+          draft = { testField: testValue };
+          store = storeWithDraft(draft);
+          testInstance = render(
+            <DraftCheckboxInput component={MockCheckboxInput} name="testField" />,
+            { store }
+          );
+        });
+
+        test('sets checked prop', () => {
+          expect(testInstance.findByType(MockCheckboxInput).props.checked).toBe(testValue);
+        });
+
+        test('calls patchDraft on change', () => {
+          const input = testInstance.findByType(MockCheckboxInput).instance;
+          input.setChecked(!testValue);
+          expect(patchDraft).toHaveBeenCalledWith({ testField: !testValue });
+        });
+      });
+    });
+  });
+});
+
+/*
```

Comment:
Again, a copy-and-paste error reflecting an earlier version of this PR. Should be removed.